cloudflare/resource_cloudflare_worker_secret_test.go

```diff
@@ -0,0 +1,165 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	cleanhttp "github.com/hashicorp/go-cleanhttp"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/cloudflare/terraform-provider-cloudflare/version"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/logging"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/httpclient"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+const scriptContentForSecret = `addEventListener('fetch', event => {event.respondWith(new Response('test 1'))});`
+
+var workerSecretTestScriptName string
+
+func TestAccCloudflareWorkerSecret_Basic(t *testing.T) {
+	t.Parallel()
+
+	name := generateRandomResourceName()
+	secretText := generateRandomResourceName()
+	workerSecretTestScriptName = generateRandomResourceName()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckAccount(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareWorkerSecretDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareWorkerSecretWithWorkerScript(workerSecretTestScriptName, name, secretText),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareWorkerSecretExists(workerSecretTestScriptName, name, secretText),
+				),
+			},
+		},
+	})
+
+	deleteWorker()
+}
+
+func testAccCheckCloudflareWorkerSecretDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*cloudflare.API)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cloudflare_worker_secret`" {
+			continue
+		}
+
+		params := getRequestParamsFromResource(rs)
+		secretResponse, err := client.ListWorkersSecrets(context.Background(), params.ScriptName)
+
+		if err != nil {
+			return err
+		}
+
+		if len(secretResponse.Result) > 0 {
+			return fmt.Errorf("Worker secret with name %s still exists against Work Script %s", secretResponse.Result[0].Name, params.ScriptName)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckCloudflareWorkerSecretWithWorkerScript(scriptName string, name string, secretText string) string {
+	err := createWorker()
```

Comment:
the way that we usually do this is create the resource as a part of the config where you include all the resources and from there, reference the values in the dependant resources. this ensures that we're using the resources under test and not using custom code (like we have here) where the codepaths may be different. check out Access resource tests for some examples of this one.